app/src/main/java/com/codingblocks/cbonlineapp/viewmodels/VideoPlayerViewModel.kt

```diff
@@ -0,0 +1,77 @@
+import com.codingblocks.cbonlineapp.database.CourseRunDao
+import com.codingblocks.cbonlineapp.database.DoubtsDao
+import com.codingblocks.cbonlineapp.database.NotesDao
+import com.codingblocks.cbonlineapp.database.models.DoubtsModel
+import com.codingblocks.cbonlineapp.database.models.NotesModel
+import com.codingblocks.cbonlineapp.extensions.retrofitCallback
+import com.codingblocks.onlineapi.Clients
+import com.codingblocks.onlineapi.models.DoubtsJsonApi
+import com.codingblocks.onlineapi.models.Notes
+import java.lang.Exception
+
+class VideoPlayerViewModel(
+    var doubtsDao: DoubtsDao,
+    var notesDao: NotesDao,
+    var courseDao: CourseDao,
+    var runDao: CourseRunDao
+) : ViewModel() {
+    var playWhenReady = false
+    var currentOrientation: Int = 0
+    var mOtp: String? = null
+    var mPlaybackInfo: String? = null
+
+    var getOtpProgress: MutableLiveData<Boolean> = MutableLiveData()
+    var createDoubtProgress: MutableLiveData<Boolean> = MutableLiveData()
+    var createNoteProgress: MutableLiveData<Boolean> = MutableLiveData()
+
+    fun getOtp(videoId: String, sectionId: String, attemptId: String) {
+        Clients.api.getOtp(videoId, sectionId, attemptId)
+            .enqueue(retrofitCallback { _, response ->
+                response?.let {
+                    mOtp = response.body()?.get("otp")?.asString
+                    mPlaybackInfo = response.body()?.get("playbackInfo")?.asString
+                    getOtpProgress.value = (it.isSuccessful)
+                }
+            })
+    }
+
+    fun createDoubt(doubt: DoubtsJsonApi) {
+        Clients.onlineV2JsonApi.createDoubt(doubt)
+            .enqueue(retrofitCallback { _, response ->
+                doubtsDao.insert(
+                    DoubtsModel(
+                        doubt.id, doubt.title, doubt.body, doubt.contents?.id
+                        ?: "", doubt.status, doubt.runAttempt?.id ?: ""
+                    )
+                )
+                createDoubtProgress.value = (response?.isSuccessful == true)
+            })
+    }
+
+    fun createNote(note: Notes) {
+        Clients.onlineV2JsonApi.createNote(note)
+            .enqueue(retrofitCallback { _, response ->
+                try {
+                    notesDao.insert(
+                        NotesModel(
+                            response?.body()?.id ?: "",
+                            response?.body()?.duration ?: 0.0,
+                            response?.body()?.text ?: "",
+                            response?.body()?.content?.id ?: "",
+                            response?.body()?.runAttempt?.id ?: "",
+                            response?.body()?.createdAt ?: "",
+                            response?.body()?.deletedAt ?: ""
+                        )
+                    )
+                } catch (e: Exception) {
+                    getOtpProgress.value = false
+                }
+                getOtpProgress.value = (response?.isSuccessful == true)
```

Comment:
otpProgress ?

Reply:
Forgot to remove it after setting observer, thanks

Reply:
Yes when response is received, we need to initialize the view. So it acts as a listener as you said

Reply:
how are notes related to otpProgress?

Reply:
Oh wait, my bad. I intialized the wrong variable. Updated, have a look